lib/strong_resources/controller/mixin.rb

```diff
@@ -1,17 +1,33 @@
 module StrongResources
   module Controller
     module Mixin
-      def self.included(klass)
-        klass.class_eval do
-          extend ClassMethods
-          class << self
-            attr_accessor :_strong_resources
-          end
+      class JSONParams
+        attr_reader :params
+        def initialize(params)
+          @params = params
+        end
+
+        def type
+          params.try(:[], :data).try(:[], :type).try(:to_sym)
         end
       end
 
+
+      extend ActiveSupport::Concern
+      included do
+        class_attribute :_strong_resources, instance_writer: false
+        self._strong_resources = {}
+      end
+
       def strong_resource
-        resource = self.class._strong_resources[action_name.to_sym]
+        params_method = if respond_to?(:raw_params) 
```

Comment:
maybe just pass the controller itself to the helper class's constructor? That way this logic is all contained in one isolated class.